src/Cake.Common/Tools/VSWhere/Product/VSWhereProduct.cs

```diff
@@ -55,6 +56,10 @@ private ProcessArgumentBuilder GetArguments(VSWhereProductSettings settings)
             {
                 builder.AppendQuoted(settings.Products);
             }
+            else
+            {
+                builder.AppendQuoted("*");
```

Comment:
Well, yes, if you specify `-products`. But if they didn't pass a value, don't specify `-products`. An argument is required only when the parameter is specified; otherwise, it defaults to:

* Microsoft.VisualStudio.Product.Community
* Microsoft.VisualStudio.Product.Professional
* Microsoft.VisualStudio.Product.Enterprise